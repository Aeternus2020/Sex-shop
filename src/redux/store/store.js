import { applyMiddleware, combineReducers, createStore } from "redux";
import products from "../getProducts/reducers";
import basket from "../pages/basket/reducers";
import favorite from "../pages/favorite/reducers";
import showModal from "../components/modal/reducers"; 
import title from "../pages/app/reducers";
import { composeEnhancers, middleware } from "./middleware";

export const rootReducer = combineReducers({
    products,
    basket,
    favorite,
    showModal,
    title
})

export const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middleware))
    )