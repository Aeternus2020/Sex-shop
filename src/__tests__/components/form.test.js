import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import Form from "../../components/form";
import { Provider } from "react-redux";
import { store } from "../../redux/store/store";
import renderer from 'react-test-renderer';

afterEach(cleanup);

test("Test Form", () => {

    render(
    <Provider store={store}>
        <Form id={3} />
    </Provider>
    );

    expect(screen.getByTestId("form-3")).toBeInTheDocument();
    expect(screen.getByTestId("form-3")).toHaveTextContent('Ordering');
    expect(screen.getByRole('button')).toHaveTextContent('Checkout');
    expect(screen.getAllByRole("textbox")).toHaveLength(4);
    fireEvent.click(screen.getByRole("button"));
});

test('Test snapshot Form', () => {
    const tree = renderer.create(
    <Provider store={store} >
        <Form id={3}/>
    </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})