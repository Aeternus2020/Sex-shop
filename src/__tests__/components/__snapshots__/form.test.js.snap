// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Form 1`] = `
<form
  className="form"
  data-testid="form-3"
  onSubmit={[Function]}
>
  <div
    className="title"
  >
    Ordering
  </div>
  <div>
    <input
      id="firstName"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="FirstName"
      type="text"
      value=""
    />
  </div>
  <div>
    <input
      id="lastName"
      name="lastName"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="LastName"
      type="text"
      value=""
    />
  </div>
  <div>
    <input
      id="age"
      name="age"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Age"
      type="number"
      value=""
    />
  </div>
  <div>
    <input
      id="address"
      name="address"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Address"
      type="text"
      value=""
    />
  </div>
  <div>
    <input
      id="phone"
      inputMode="numeric"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      type="text"
      value="+_ (___) ____ ___"
    />
  </div>
  <button
    type="submit"
  >
    Checkout
  </button>
</form>
`;
